### DISCUSSION TITLE
error: removeSubscription is not a function

### DISCUSSION THREAD

**Discussion by felibunnyy** [supabase/supabase#7844] [Questions] [answered]
I am currently trying to establish a realtime subscription for supabase in a useEffect hook, this is the code for the cleanup function that I have that seems to throw an error. Anyone facing the same issues/has any idea how to debug this error?

The subscription seems to be working the first time the page renders, but for subsequent attempts, the subscription has property of "state: closed". 

![Screenshot 2022-07-22 at 2 39 28 PM](https://user-images.githubusercontent.com/105587981/180378917-03ef8deb-f9ad-4010-98f0-79357c4db6c2.png)

this is the screenshot of the error as shown in console:
![Screenshot 2022-07-22 at 2 36 59 PM](https://user-images.githubusercontent.com/105587981/180378632-5fc5bb74-a299-4772-a19c-20ab3379af54.png)

**Comment by GaryAustin1** [comment, marked as answer]
What is .removeMessageSubscription? Is that documented somewhere.  

Also react strict mode seems to be causing issues.  https://github.com/supabase/realtime-js/issues/169
